UICatalog/Scenarios/VirtualContent.cs

```diff
@@ -0,0 +1,343 @@
+
+        void AllowNegativeY_Toggled (object sender, StateEventArgs<bool?> e)
+        {
+            if (e.NewValue == true)
+            {
+                view.ViewportSettings |= ViewportSettings.AllowNegativeY;
+            }
+            else
+            {
+                view.ViewportSettings &= ~ViewportSettings.AllowNegativeY;
+            }
+        }
+
+        view.Padding.Add (cbAllowNegativeY);
+
+        var cbAllowXGreaterThanContentWidth = new CheckBox
+        {
+            Title = "Allow X > Content",
+            Y = Pos.Bottom (cbAllowNegativeX),
+            CanFocus = false
+        };
+        cbAllowXGreaterThanContentWidth.Checked = view.ViewportSettings.HasFlag (ViewportSettings.AllowXGreaterThanContentWidth);
+        cbAllowXGreaterThanContentWidth.Toggled += AllowXGreaterThanContentWidth_Toggled;
+
+        void AllowXGreaterThanContentWidth_Toggled (object sender, StateEventArgs<bool?> e)
+        {
+            if (e.NewValue == true)
+            {
+                view.ViewportSettings |= ViewportSettings.AllowXGreaterThanContentWidth;
+            }
+            else
+            {
+                view.ViewportSettings &= ~ViewportSettings.AllowXGreaterThanContentWidth;
+            }
+        }
+
+        view.Padding.Add (cbAllowXGreaterThanContentWidth);
+
+        var cbAllowYGreaterThanContentHeight = new CheckBox
+        {
+            Title = "Allow Y > Content",
+            X = Pos.Right (cbAllowXGreaterThanContentWidth) + 1,
+            Y = Pos.Bottom (cbAllowNegativeX),
+            CanFocus = false
+        };
+        cbAllowYGreaterThanContentHeight.Checked = view.ViewportSettings.HasFlag (ViewportSettings.AllowYGreaterThanContentHeight);
+        cbAllowYGreaterThanContentHeight.Toggled += AllowYGreaterThanContentHeight_Toggled;
+
+        void AllowYGreaterThanContentHeight_Toggled (object sender, StateEventArgs<bool?> e)
+        {
+            if (e.NewValue == true)
+            {
+                view.ViewportSettings |= ViewportSettings.AllowYGreaterThanContentHeight;
+            }
+            else
+            {
+                view.ViewportSettings &= ~ViewportSettings.AllowYGreaterThanContentHeight;
+            }
+        }
+
+        view.Padding.Add (cbAllowYGreaterThanContentHeight);
+
+        var labelContentSize = new Label
+        {
+            Title = "_ContentSize:",
+            Y = Pos.Bottom (cbAllowYGreaterThanContentHeight)
+        };
+
+        var contentSizeWidth = new Buttons.NumericUpDown
+        {
+            Value = view.ContentSize.Width,
+            X = Pos.Right (labelContentSize) + 1,
+            Y = Pos.Top (labelContentSize)
+        };
+        contentSizeWidth.ValueChanged += ContentSizeWidth_ValueChanged;
+
+        void ContentSizeWidth_ValueChanged (object sender, PropertyChangedEventArgs e)
+        {
+            view.ContentSize = view.ContentSize with { Width = ((Buttons.NumericUpDown)sender).Value };
```

Comment:
Tell me more about what you mean by "inverting vierws"? Can you describe a real-world-example where  this would be useful? I have an intuition that you're on to something, but I'm strugging to come up with an example myself.

Reply:
> Tell me more about what you mean by "inverting vierws"? Can you describe a real-world-example where this would be useful? I have an intuition that you're on to something, but I'm strugging to come up with an example myself.

I just tried to find reversed letters and characters to put my idea into practice, but I only found letters from top to bottom and none from left to right. Like simulating the mirror effect but in a TUI this doesn't seem possible, I think.